src/test/java/api/loans/CheckOutByBarcodeTests.java

```diff
@@ -2567,6 +2567,35 @@ void shouldNotFailIfPatronDoesNotHaveBarcodeAndItemWasAlreadyRequested() {
       hasUserBarcodeParameter(steveWithNoBarcode))));
   }
 
+  @Test
+  void checkOutByBarcodeWithCheckOutLockFeatureEnabled() {
+    IndividualResource item1 = itemsFixture.basedUponNod();
+    IndividualResource item2 = itemsFixture.basedUponSmallAngryPlanet();
+    IndividualResource item3 = itemsFixture.basedUponTemeraire();
+    IndividualResource item4 = itemsFixture.basedUponDunkirk();
+    IndividualResource rebecca = usersFixture.rebecca();
+
+    //Normal checkout without checkOutLockFeature Enabled
+    checkOutFixture.checkOutByBarcode(item1, rebecca);
+
+    //Enabling checkOutLockFeature
+    settingsFixture.enableCheckoutLockFeature(true);
+    //Lock creation and deletion will happen in the same checkout
+    checkOutFixture.checkOutByBarcode(item2, rebecca);
+
+    //Creating a lock for user so that user will not be able to acquire lock
+    IndividualResource checkOutLock = checkOutLockFixture.createLockForUserId(rebecca.getId().toString());
+    final Response response = checkOutFixture.attemptCheckOutByBarcode(item3, rebecca);
+    assertThat(response, hasStatus(HTTP_UNPROCESSABLE_ENTITY));
+    assertThat(response.getJson(),
+      hasErrorWith(hasMessage("unable to acquire lock")));
+
+    //Deleting the lock
+    checkOutLockFixture.deleteLock(checkOutLock.getId());
+
+    checkOutFixture.checkOutByBarcode(item4, rebecca);
+  }
+
```

Comment:
Do we need a test that `validateItemLimitBasedOnLockFeatureFlag` works as expected? Maybe a test fixture that sets the limit, checks out multiple items, but goes over over the limit to cause the error.

Reply:
This feature can only be tested by triggering checkouts concurrently. This scenario is implemented in Karate test and Please find the PR below.
**https://github.com/folio-org/folio-integration-tests/pull/1002**

Reply:
Hi @steveellis , the test case to test the concurrent checkout is added.